### PR TITLE
Clean up API: rename story -> stories, add basic error handling

### PR DESCRIPTION
**Pull request by davesteinberg**
@mwickett It seems I don't actually have permission to push to a new branch, so I've created this PR against a fork. I've rebased, so this should be a clean merge on your last commit. It'd be helpful if you could merge this before it goes stale again.

This just cleans up the APIs a little bit:

- Moves `/api/story` → `/api/stories` (conventional for a collection)
- Excludes sensitive fields from GET results
- Handles some obvious errors (input validation is just DB validation, not great)
- Uses status 201 for Created
- Eliminates "API resolved without sending a response" Next errors
- Eliminates lint warnings

It doesn't add anything functional to the app at this point, but it was helpful to play around with Next API routes (a refresher) and Prisma (new to me). I'll probably continue to refactor as I work on the stories list.

### CODE REVIEW

**Comment by mwickett** [comment]
ah, @davesteinberg - sorry, you're added with write permissions now!

**Review by mwickett — APPROVED** [review]
🎸 Great improvements here.

**Comment by davesteinberg** [comment]
@mwickett Thanks for the approval. I still don't have write access, though. I can't push a branch and I can't merge this PR.

**Pull request merged**
